dora/core/common/src/main/java/alluxio/membership/EtcdMembershipManager.java

```diff
@@ -116,11 +118,17 @@ public void join(WorkerInfo workerInfo) throws IOException {
 
   @Override
   public WorkerClusterView getAllMembers() throws IOException {
+    Set<WorkerIdentity> liveWorkerIds = parseWorkersFromEtcdKvPairs(
+        mAlluxioEtcdClient.mServiceDiscovery.getAllLiveServices())
+        .map(WorkerServiceEntity::getIdentity)
+        .collect(Collectors.toSet());
+    Predicate<WorkerInfo> isLive = w -> liveWorkerIds.contains(w.getIdentity());
```

Comment:
to figure out the liveness of a worker, I had to first get a set of all live workers, so that live and lost workers can be correctly differentiate. @lucyge2022 I'd appreciate your comment on this.

Reply:
yeah u have to make separate calls to know the state of a worker